core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -452,19 +452,20 @@ object StorageTool extends Logging {
       stream.println("All of the log directories are already formatted.")
     } else {
       metaPropertiesEnsemble.emptyLogDirs().forEach(logDir => {
-        copier.setLogDirProps(logDir, new MetaProperties.Builder(metaProperties).
-          setDirectoryId(copier.generateValidDirectoryId()).
+        val loggingCopier = new MetaPropertiesEnsemble.Copier(metaPropertiesEnsemble)
+        loggingCopier.setLogDirProps(logDir, new MetaProperties.Builder(metaProperties).
+          setDirectoryId(loggingCopier.generateValidDirectoryId()).
           build())
-        copier.setPreWriteHandler((logDir, _, _) => {
+        loggingCopier.setPreWriteHandler((logDir, _, _) => {
           stream.println(s"Formatting $logDir with metadata.version $metadataVersion.")
           Files.createDirectories(Paths.get(logDir))
           val bootstrapDirectory = new BootstrapDirectory(logDir, Optional.empty())
           bootstrapDirectory.writeBinaryFile(bootstrapMetadata)
         })
-        copier.setWriteErrorHandler((logDir, e) => {
+        loggingCopier.setWriteErrorHandler((logDir, e) => {
           throw new TerseFailure(s"Error while writing meta.properties file $logDir: ${e.getMessage}")
         })
-        copier.writeLogDirChanges()
```

Comment:
Maybe the simpler solution is move `copier.writeLogDirChanges()` out of loop.

Reply:
If just move this line out of loop,print in reverse order.
The test cases have been added. please review @chia7712

Reply:
If we do care the order of format, we should change `logDirProps` from `HashMap` to `LinkedHashMap`

https://github.com/apache/kafka/blob/5552f5c26df4eb07b2d6ee218e4a29e4ca790d5c/metadata/src/main/java/org/apache/kafka/metadata/properties/MetaPropertiesEnsemble.java#L192

However, I don't think the order is important, since all we want to do is to make sure all of them get format only once, right?